checker/src/main/java/org/checkerframework/checker/calledmethods/CalledMethodsAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,406 @@
+
+/** The annotated type factory for the Called Methods checker. */
+public class CalledMethodsAnnotatedTypeFactory extends AccumulationAnnotatedTypeFactory {
+
+    /**
+     * The {@link java.util.Collections#singletonList} method. It is treated specially by {@link
+     * #adjustMethodNameUsingValueChecker(String, MethodInvocationTree)}.
+     */
+    private final ExecutableElement collectionsSingletonList;
+
+    /**
+     * Whether to use the Value Checker as a subchecker to reduce false positives when analyzing
+     * calls to the AWS SDK. Defaults to false. Controlled by the command-line option {@code
+     * -AuseValueChecker}.
+     */
+    private final boolean useValueChecker;
+
+    /**
+     * The builder frameworks (such as Lombok and AutoValue) supported by the Called Methods
+     * checker.
+     */
+    private Collection<BuilderFrameworkSupport> builderFrameworkSupports;
+
+    /**
+     * Lombok has a flag to generate @CalledMethods annotations, but they used the old package name,
+     * so we maintain it as an alias.
+     */
+    private static final String OLD_CALLED_METHODS =
+            "org.checkerframework.checker.builder.qual.CalledMethods";
+
+    /**
+     * Lombok also generates an @NotCalledMethods annotation, which we have no support for. We
+     * therefore treat it as top.
+     */
+    private static final String OLD_NOT_CALLED_METHODS =
+            "org.checkerframework.checker.builder.qual.NotCalledMethods";
+
+    /**
+     * Create a new CalledMethodsAnnotatedTypeFactory.
+     *
+     * @param checker the checker
+     */
+    public CalledMethodsAnnotatedTypeFactory(BaseTypeChecker checker) {
+        super(
+                checker,
+                CalledMethods.class,
+                CalledMethodsBottom.class,
+                CalledMethodsPredicate.class);
+        Set<String> disabledFrameworks = new HashSet<>();
+        if (checker.hasOption(CalledMethodsChecker.DISABLE_BUILDER_FRAMEWORK_SUPPORTS)) {
+            disabledFrameworks.addAll(
+                    Arrays.asList(
+                                    checker.getOption(
+                                                    CalledMethodsChecker
+                                                            .DISABLE_BUILDER_FRAMEWORK_SUPPORTS)
+                                            .split(","))
+                            .stream()
+                            .map(String::toUpperCase)
+                            .collect(Collectors.toList()));
+        }
+        builderFrameworkSupports = new ArrayList<>();
+        enableFramework(CalledMethodsChecker.LOMBOK_SUPPORT, disabledFrameworks);
+        enableFramework(CalledMethodsChecker.AUTOVALUE_SUPPORT, disabledFrameworks);
+
+        if (!disabledFrameworks.isEmpty()) {
+            StringJoiner sj = new StringJoiner(", ");
+            disabledFrameworks.iterator().forEachRemaining(s -> sj.add(s));
+            String unrecognized = sj.toString();
```

Comment:
Rather than these 3 lines of code, call [`UtilPlume.join`](http://plumelib.org/plume-util/api/org/plumelib/util/UtilPlume.html#join-java.lang.CharSequence-java.lang.Iterable-).
Actually, it would be fine to just issue an error about the first incorrect string found.  That would lead to simpler code, and it is adequate for users.